pkgs/tools/security/john/default.nix

```diff
@@ -44,13 +62,42 @@ stdenv.mkDerivation rec {
     "--with-systemwide"
   ];
 
-  buildInputs = [ openssl nss nspr libkrb5 gmp zlib libpcap re2 opencl-headers ocl-icd ];
-  nativeBuildInputs = [ gcc python3Packages.wrapPython perl makeWrapper ];
-  propagatedBuildInputs = (with python3Packages; [ dpkt scapy lxml ]) ++ # For pcap2john.py
-                          (with perlPackages; [ DigestMD4 DigestSHA1 GetoptLong # For pass_gen.pl
-                                                CompressRawLzma # For 7z2john.pl
-                                                perlldap ]); # For sha-dump.pl
-                          # TODO: Get dependencies for radius2john.pl and lion2john-alt.pl
+  buildInputs = [
+    openssl
+    nss
+    nspr
+    libkrb5
+    gmp
+    zlib
+    libpcap
+    re2
+    opencl-headers
```

Comment:
OpenCL stuff should go to the other commit

Reply:
Sorry, I'm not sure what you mean. Right now there are three commits:

1. Add OpenCL
2. Reformat
3. Make OpenCL optional

Would you like me to reorder / merge some of these?

Reply:
Squash the OpenCL commits.

It should be:
1. reformat
2. Add OpenCL support

Reply:
OK, done.